trino/client.py

```diff
@@ -758,7 +676,14 @@ def execute(self, additional_http_headers=None) -> TrinoResult:
         self._warnings = getattr(status, "warnings", [])
         if status.next_uri is None:
             self._finished = True
-        self._result = TrinoResult(self, status.rows, self._experimental_python_types)
+
+        if self._experimental_python_types:
+            factory = RowMapperFactory(status.columns)
+            rows = factory.map_rows(status.rows)
+        else:
+            rows = status.rows
+
+        self._result = TrinoResult(self, rows)
```

Comment:
Also applies to the `fetch` method.

```suggestion
        assert self._row_mapper is not None
        self._result = TrinoResult(self, self._row_mapper(rows))
```

Reply:
Actually if we want to call `RowMapperFactory()` only once we know the column, then we shouldn't call this in `execute()`. At this point we have no guarantee we'll have columns at this point.

Reply:
Correct, that's why I suggested to add this in the `_update_state` method to call it only when you actually have the columns.

Reply:
Done